src/sentry/static/sentry/app/components/resultGrid.tsx

```diff
@@ -80,36 +83,46 @@ class Filter extends React.Component {
   }
 }
 
-class SortBy extends React.Component {
+type SortByProps = {
+  options: Option[];
+  path: string;
+  location: Location;
+  value: string;
+};
+
+class SortBy extends React.Component<SortByProps> {
   static propTypes = {
     options: PropTypes.array.isRequired,
     path: PropTypes.string.isRequired,
     location: PropTypes.object,
     value: PropTypes.any,
   };
 
-  getCurrentSortLabel = () =>
-    this.props.options.filter(item => item[0] === this.props.value)[0][1];
+  getCurrentSortLabel() {
+    return this.props.options.filter(item => item[0] === this.props.value)[0][1];
+  }
 
-  getSortBySelector = () => (
-    <DropdownLink title={this.getCurrentSortLabel()} className="sorted-by">
-      {this.props.options.map(item => {
-        const query = $.extend({}, this.props.location.query, {
-          sortBy: item[0],
-          cursor: '',
-        });
-        return (
-          <MenuItem
-            isActive={this.props.value === item[0]}
-            key={item[0]}
-            to={{pathname: this.props.path, query}}
-          >
-            {item[1]}
-          </MenuItem>
-        );
-      })}
-    </DropdownLink>
-  );
+  getSortBySelector() {
+    return (
+      <DropdownLink title={this.getCurrentSortLabel()} className="sorted-by">
+        {this.props.options.map(item => {
```

Comment:
If we named the tuples, it might be nice to destructure the map callback arg so we can give them more specific names instead of referencing by index.